libs/angular/src/lib/pagination/pagination.component.ts

```diff
@@ -71,6 +71,10 @@ export class NggPaginationComponent implements OnChanges {
     ) {
       this._pageIndicies = this.getDisplayedPageIndicies()
     }
+
+    if (changes.length && this.pageIndex >= this.totalPages) {
+      this.goto(this.totalPages - 1)
```

Comment:
it looks like someone is hiding real problems, pagination component receives `pageIndex` and pages count `length`,  in my opinion this component shouldn't be responsible for wrong data. After this  - page change event is thrown meaning, that any its listeners would for example trigger endpoint call, that could lead to unexpected behaviour.

Reply:
Valid point. If the length changes you should also have calculated and change pageIndex.

I'm also thinking about what causes the length to change? Filtering? Choosing a new data source? In these cases wouldn't you want to end up on the first page? Seems like a UX problem more than a component one.

Reply:
The cause is deletion, but definitely a good point to handle it on the consumer side. Closing the PR. Thanks for the review!